### PR TITLE
Enable jndi

### LINKED ISSUE #116: Unable to use JGroups DNS_PING protocol
We rely on DNS_PING in k8s environments to discover cluster members. However, the current  (14.0.1.Final) native image does not have JNDI enabled so this always fails with the following logs:

```
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2022-10-18 10:49:36,631 INFO  [io.quarkus] (main) infinispan-quarkus-server-runner 14.0.1.Final native (powered by Quarkus 2.13.0.Final) started in 0.010s. 
2022-10-18 10:49:36,632 INFO  [io.quarkus] (main) Profile prod activated. 
2022-10-18 10:49:36,632 INFO  [io.quarkus] (main) Installed features: [cdi, infinispan-embedded, infinispan-server]
2022-10-18 10:49:36,632 INFO  [BOOT] (main) JVM Substrate VM Oracle Corporation GraalVM 22.1.0 Java 11 CE
2022-10-18 10:49:36,632 INFO  [BOOT] (main) JVM arguments = [-Dinfinispan.server.home.path=/opt/infinispan, --bind-address=0.0.0.0, -l, /opt/infinispan/server/conf/operator/log4j.xml, -c, operator/infinispan-base.xml, -c, operator/infinispan-admin.xml]
2022-10-18 10:49:36,632 INFO  [BOOT] (main) PID = 47
2022-10-18 10:49:36,632 INFO  [org.inf.SERVER] (main) ISPN080000: Infinispan Server 14.0.1.Final starting
2022-10-18 10:49:36,632 INFO  [org.inf.SERVER] (main) ISPN080017: Server configuration: operator/infinispan-base.xml
2022-10-18 10:49:36,632 INFO  [org.inf.SERVER] (main) ISPN080017: Server configuration: operator/infinispan-admin.xml
2022-10-18 10:49:36,632 INFO  [org.inf.SERVER] (main) ISPN080032: Logging configuration: /opt/infinispan/server/conf/operator/log4j.xml
2022-10-18 10:49:36,639 WARN  [org.inf.SERVER] (main) ISPN080059: No script engines are available
2022-10-18 10:49:36,642 INFO  [org.inf.CONTAINER] (main) ISPN000556: Starting user marshaller 'org.infinispan.commons.marshall.ImmutableProtoStreamMarshaller'
2022-10-18 10:49:36,651 INFO  [org.inf.CLUSTER] (main) ISPN000078: Starting JGroups channel `test-base-functionality` with stack `image-tcp`
2022-10-18 10:49:36,656 ERROR [org.inf.CONFIG] (main) ISPN000660: DefaultCacheManager start failed, stopping any running components: org.infinispan.commons.CacheConfigurationException: ISPN000541: Error while trying to create a channel using the specified configuration '[TCP(bundler.max_size=64000, sock_conn_timeout=300, thread_pool.keep_alive_time=60000, port_range=0, diag.enabled=false, bind_port=7800, thread_naming_pattern=pl, thread_pool.thread_dumps_threshold=10000, send_buf_size=640k, thread_pool.max_threads=200, bundler_type=transfer-queue, bind_addr=SITE_LOCAL, thread_pool.min_threads=0), RED(), MPING(ip_ttl=2, mcast_port=46655, mcast_addr=239.6.7.8, num_discovery_runs=3), MERGE3(max_interval=30000, min_interval=10000), FD_SOCK2(offset=50000), FD_ALL3(), VERIFY_SUSPECT2(timeout=1000), pbcast.NAKACK2(xmit_table_num_rows=50, use_mcast_xmit=false, xmit_table_msgs_per_row=1024, xmit_table_max_compaction_time=30000, xmit_interval=200, resend_last_seqno=true), UNICAST3(conn_close_timeout=5000, xmit_interval=200, xmit_table_num_rows=50, xmit_table_msgs_per_row=1024, xmit_table_max_compaction_time=30000), pbcast.STABLE(desired_avg_gossip=5000, max_bytes=1M), pbcast.GMS(join_timeout=2000, print_local_addr=false), UFC(min_threshold=0.40, max_credits=4m), MFC(min_threshold=0.40, max_credits=4m), FRAG4(frag_size=60000), dns.DNS_PING(dns_record_type=A, dns_query=test-base-functionality-ping.namespace-for-testing.svc.cluster.local)]'
	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.channelFromConfigurator(JGroupsTransport.java:776)
	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.buildChannel(JGroupsTransport.java:745)
	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.initChannel(JGroupsTransport.java:510)
	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.start(JGroupsTransport.java:491)
	at org.infinispan.remoting.transport.jgroups.CorePackageImpl$1.start(CorePackageImpl.java:42)
	at org.infinispan.remoting.transport.jgroups.CorePackageImpl$1.start(CorePackageImpl.java:27)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.invokeStart(BasicComponentRegistryImpl.java:617)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.doStartWrapper(BasicComponentRegistryImpl.java:608)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.startWrapper(BasicComponentRegistryImpl.java:577)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl$ComponentWrapper.running(BasicComponentRegistryImpl.java:808)
	at org.infinispan.metrics.impl.MetricsCollector.start(MetricsCollector.java:76)
	at org.infinispan.metrics.impl.CorePackageImpl$1.start(CorePackageImpl.java:41)
	at org.infinispan.metrics.impl.CorePackageImpl$1.start(CorePackageImpl.java:34)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.invokeStart(BasicComponentRegistryImpl.java:617)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.doStartWrapper(BasicComponentRegistryImpl.java:608)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.startWrapper(BasicComponentRegistryImpl.java:577)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl$ComponentWrapper.running(BasicComponentRegistryImpl.java:808)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.startDependencies(BasicComponentRegistryImpl.java:635)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.doStartWrapper(BasicComponentRegistryImpl.java:599)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.startWrapper(BasicComponentRegistryImpl.java:577)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl$ComponentWrapper.running(BasicComponentRegistryImpl.java:808)
	at org.infinispan.factories.AbstractComponentRegistry.internalStart(AbstractComponentRegistry.java:357)
	at org.infinispan.factories.AbstractComponentRegistry.start(AbstractComponentRegistry.java:250)
	at org.infinispan.manager.DefaultCacheManager.internalStart(DefaultCacheManager.java:774)
	at org.infinispan.manager.DefaultCacheManager.start(DefaultCacheManager.java:742)
	at org.infinispan.server.SecurityActions.lambda$startCacheManager$1(SecurityActions.java:68)
	at org.infinispan.security.Security.doPrivileged(Security.java:56)
	at org.infinispan.server.SecurityActions.doPrivileged(SecurityActions.java:40)
	at org.infinispan.server.SecurityActions.startCacheManager(SecurityActions.java:71)
	at org.infinispan.server.Server.run(Server.java:408)
	at org.infinispan.server.Bootstrap.runInternal(Bootstrap.java:173)
	at org.infinispan.server.tool.Main.run(Main.java:98)
	at org.infinispan.server.Bootstrap.main(Bootstrap.java:56)
	at org.infininspan.quarkus.server.InfinispanQuarkusServer.run(InfinispanQuarkusServer.java:13)
	at org.infininspan.quarkus.server.InfinispanQuarkusServer_ClientProxy.run(Unknown Source)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:131)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:69)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:42)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
Caused by: javax.naming.NamingException: JNDI has been disabled, enable it with quarkus.naming.enable-jndi=true
	at io.quarkus.runtime.naming.DisabledInitialContext.init(DisabledInitialContext.java:210)
	at javax.naming.InitialContext.<init>(InitialContext.java:208)
	at javax.naming.directory.InitialDirContext.<init>(InitialDirContext.java:101)
	at javax.naming.ldap.InitialLdapContext.<init>(InitialLdapContext.java:102)
	at io.quarkus.runtime.naming.DisabledInitialContext.<init>(DisabledInitialContext.java:32)
	at io.quarkus.runtime.naming.DisabledInitialContextManager$1.getInitialContext(DisabledInitialContextManager.java:28)
	at javax.naming.spi.NamingManager.getInitialContext(NamingManager.java:730)
	at javax.naming.InitialContext.getDefaultInitCtx(InitialContext.java:305)
	at javax.naming.InitialContext.init(InitialContext.java:236)
	at javax.naming.InitialContext.<init>(InitialContext.java:208)
	at javax.naming.directory.InitialDirContext.<init>(InitialDirContext.java:101)
	at org.jgroups.protocols.dns.DefaultDNSResolver.<init>(DefaultDNSResolver.java:42)
	at org.jgroups.protocols.dns.DNS_PING.init(DNS_PING.java:58)
	at org.jgroups.stack.ProtocolStack.initProtocolStack(ProtocolStack.java:794)
	at org.jgroups.stack.ProtocolStack.initProtocolStack(ProtocolStack.java:781)
	at org.jgroups.stack.ProtocolStack.init(ProtocolStack.java:777)
	at org.jgroups.JChannel.<init>(JChannel.java:155)
	at org.infinispan.remoting.transport.jgroups.EmbeddedJGroupsChannelConfigurator.createChannel(EmbeddedJGroupsChannelConfigurator.java:136)
	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.channelFromConfigurator(JGroupsTransport.java:774)
	... 38 more

2022-10-18 10:49:36,656 WARN  [org.inf.CONTAINER] (main) ISPN000574: Global state cannot persisted because it is incomplete (usually caused by errors at startup).
2022-10-18 10:49:36,758 FATAL [org.inf.SERVER] (main) ISPN080028: Infinispan Server failed to start: org.infinispan.manager.EmbeddedCacheManagerStartupException: org.infinispan.commons.CacheConfigurationException: ISPN000541: Error while trying to create a channel using the specified configuration '[TCP(bundler.max_size=64000, sock_conn_timeout=300, thread_pool.keep_alive_time=60000, port_range=0, diag.enabled=false, bind_port=7800, thread_naming_pattern=pl, thread_pool.thread_dumps_threshold=10000, send_buf_size=640k, thread_pool.max_threads=200, bundler_type=transfer-queue, bind_addr=SITE_LOCAL, thread_pool.min_threads=0), RED(), MPING(ip_ttl=2, mcast_port=46655, mcast_addr=239.6.7.8, num_discovery_runs=3), MERGE3(max_interval=30000, min_interval=10000), FD_SOCK2(offset=50000), FD_ALL3(), VERIFY_SUSPECT2(timeout=1000), pbcast.NAKACK2(xmit_table_num_rows=50, use_mcast_xmit=false, xmit_table_msgs_per_row=1024, xmit_table_max_compaction_time=30000, xmit_interval=200, resend_last_seqno=true), UNICAST3(conn_close_timeout=5000, xmit_interval=200, xmit_table_num_rows=50, xmit_table_msgs_per_row=1024, xmit_table_max_compaction_time=30000), pbcast.STABLE(desired_avg_gossip=5000, max_bytes=1M), pbcast.GMS(join_timeout=2000, print_local_addr=false), UFC(min_threshold=0.40, max_credits=4m), MFC(min_threshold=0.40, max_credits=4m), FRAG4(frag_size=60000), dns.DNS_PING(dns_record_type=A, dns_query=test-base-functionality-ping.namespace-for-testing.svc.cluster.local)]'
	at org.infinispan.manager.DefaultCacheManager.internalStart(DefaultCacheManager.java:778)
	at org.infinispan.manager.DefaultCacheManager.start(DefaultCacheManager.java:742)
	at org.infinispan.server.SecurityActions.lambda$startCacheManager$1(SecurityActions.java:68)
	at org.infinispan.security.Security.doPrivileged(Security.java:56)
	at org.infinispan.server.SecurityActions.doPrivileged(SecurityActions.java:40)
	at org.infinispan.server.SecurityActions.startCacheManager(SecurityActions.java:71)
	at org.infinispan.server.Server.run(Server.java:408)
	at org.infinispan.server.Bootstrap.runInternal(Bootstrap.java:173)
	at org.infinispan.server.tool.Main.run(Main.java:98)
	at org.infinispan.server.Bootstrap.main(Bootstrap.java:56)
	at org.infininspan.quarkus.server.InfinispanQuarkusServer.run(InfinispanQuarkusServer.java:13)
	at org.infininspan.quarkus.server.InfinispanQuarkusServer_ClientProxy.run(Unknown Source)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:131)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:69)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:42)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
Caused by: org.infinispan.commons.CacheConfigurationException: ISPN000541: Error while trying to create a channel using the specified configuration '[TCP(bundler.max_size=64000, sock_conn_timeout=300, thread_pool.keep_alive_time=60000, port_range=0, diag.enabled=false, bind_port=7800, thread_naming_pattern=pl, thread_pool.thread_dumps_threshold=10000, send_buf_size=640k, thread_pool.max_threads=200, bundler_type=transfer-queue, bind_addr=SITE_LOCAL, thread_pool.min_threads=0), RED(), MPING(ip_ttl=2, mcast_port=46655, mcast_addr=239.6.7.8, num_discovery_runs=3), MERGE3(max_interval=30000, min_interval=10000), FD_SOCK2(offset=50000), FD_ALL3(), VERIFY_SUSPECT2(timeout=1000), pbcast.NAKACK2(xmit_table_num_rows=50, use_mcast_xmit=false, xmit_table_msgs_per_row=1024, xmit_table_max_compaction_time=30000, xmit_interval=200, resend_last_seqno=true), UNICAST3(conn_close_timeout=5000, xmit_interval=200, xmit_table_num_rows=50, xmit_table_msgs_per_row=1024, xmit_table_max_compaction_time=30000), pbcast.STABLE(desired_avg_gossip=5000, max_bytes=1M), pbcast.GMS(join_timeout=2000, print_local_addr=false), UFC(min_threshold=0.40, max_credits=4m), MFC(min_threshold=0.40, max_credits=4m), FRAG4(frag_size=60000), dns.DNS_PING(dns_record_type=A, dns_query=test-base-functionality-ping.namespace-for-testing.svc.cluster.local)]'
	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.channelFromConfigurator(JGroupsTransport.java:776)
	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.buildChannel(JGroupsTransport.java:745)
	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.initChannel(JGroupsTransport.java:510)
	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.start(JGroupsTransport.java:491)
	at org.infinispan.remoting.transport.jgroups.CorePackageImpl$1.start(CorePackageImpl.java:42)
	at org.infinispan.remoting.transport.jgroups.CorePackageImpl$1.start(CorePackageImpl.java:27)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.invokeStart(BasicComponentRegistryImpl.java:617)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.doStartWrapper(BasicComponentRegistryImpl.java:608)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.startWrapper(BasicComponentRegistryImpl.java:577)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl$ComponentWrapper.running(BasicComponentRegistryImpl.java:808)
	at org.infinispan.metrics.impl.MetricsCollector.start(MetricsCollector.java:76)
	at org.infinispan.metrics.impl.CorePackageImpl$1.start(CorePackageImpl.java:41)
	at org.infinispan.metrics.impl.CorePackageImpl$1.start(CorePackageImpl.java:34)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.invokeStart(BasicComponentRegistryImpl.java:617)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.doStartWrapper(BasicComponentRegistryImpl.java:608)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.startWrapper(BasicComponentRegistryImpl.java:577)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl$ComponentWrapper.running(BasicComponentRegistryImpl.java:808)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.startDependencies(BasicComponentRegistryImpl.java:635)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.doStartWrapper(BasicComponentRegistryImpl.java:599)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.startWrapper(BasicComponentRegistryImpl.java:577)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl$ComponentWrapper.running(BasicComponentRegistryImpl.java:808)
	at org.infinispan.factories.AbstractComponentRegistry.internalStart(AbstractComponentRegistry.java:357)
	at org.infinispan.factories.AbstractComponentRegistry.start(AbstractComponentRegistry.java:250)
	at org.infinispan.manager.DefaultCacheManager.internalStart(DefaultCacheManager.java:774)
	... 15 more
Caused by: javax.naming.NamingException: JNDI has been disabled, enable it with quarkus.naming.enable-jndi=true
	at io.quarkus.runtime.naming.DisabledInitialContext.init(DisabledInitialContext.java:210)
	at javax.naming.InitialContext.<init>(InitialContext.java:208)
	at javax.naming.directory.InitialDirContext.<init>(InitialDirContext.java:101)
	at javax.naming.ldap.InitialLdapContext.<init>(InitialLdapContext.java:102)
	at io.quarkus.runtime.naming.DisabledInitialContext.<init>(DisabledInitialContext.java:32)
	at io.quarkus.runtime.naming.DisabledInitialContextManager$1.getInitialContext(DisabledInitialContextManager.java:28)
	at javax.naming.spi.NamingManager.getInitialContext(NamingManager.java:730)
	at javax.naming.InitialContext.getDefaultInitCtx(InitialContext.java:305)
	at javax.naming.InitialContext.init(InitialContext.java:236)
	at javax.naming.InitialContext.<init>(InitialContext.java:208)
	at javax.naming.directory.InitialDirContext.<init>(InitialDirContext.java:101)
	at org.jgroups.protocols.dns.DefaultDNSResolver.<init>(DefaultDNSResolver.java:42)
	at org.jgroups.protocols.dns.DNS_PING.init(DNS_PING.java:58)
	at org.jgroups.stack.ProtocolStack.initProtocolStack(ProtocolStack.java:794)
	at org.jgroups.stack.ProtocolStack.initProtocolStack(ProtocolStack.java:781)
	at org.jgroups.stack.ProtocolStack.init(ProtocolStack.java:777)
	at org.jgroups.JChannel.<init>(JChannel.java:155)
	at org.infinispan.remoting.transport.jgroups.EmbeddedJGroupsChannelConfigurator.createChannel(EmbeddedJGroupsChannelConfigurator.java:136)
	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.channelFromConfigurator(JGroupsTransport.java:774)
	... 38 more

2022-10-18 10:49:36,759 INFO  [io.quarkus] (main) infinispan-quarkus-server-runner stopped in 0.000s
```

Adding the `quarkus.naming.enable-jndi=true` option at build time could resolve the issue.

### PR DESCRIPTION
**Pull request by ryanemerson**
Resolves #116

### CODE REVIEW

File: server/runtime/src/main/java/org/infinispan/quarkus/server/runtime/graal/SubstituteDnsLookup.java [hunk 1]
```diff
@@ -39,4 +40,15 @@ public InitialContextFactory createInitialContextFactory(Hashtable<?, ?> environ
             throw new NamingException("Native Infinispan Server does not support " + className + " as an InitialContextFactory");
       }
    }
-}
\ No newline at end of file
+}
+
+@TargetClass(NamingManager.class)
+final class Target_NamingManager{
```
Comment (wburns) [inline server/runtime/src/main/java/org/infinispan/quarkus/server/runtime/graal/SubstituteDnsLookup.java:46]: Nitpick: 
```suggestion
final class Target_NamingManager {
```

**Review by wburns — APPROVED** [review]
LGTM

**Comment by infinispanrelease** [comment]
Image pushed for Jenkins build [#2](https://ci.infinispan.org/job/Infinispan%20Quarkus/job/PR-118/2/):
```
quay.io/infinispan-test/server-native:PR-118
```

**Comment by ahayworth** [comment]
👋 Hi! I'm trying out Infinispan, and the native image looked very interesting to me as an option - I don't personally have a ton of experience with the JVM 🙇 

However, I'm running in kubernetes, and ran into this precise issues - I was wondering if this fix was going to be released soon? I'm not sure what the release schedules are like for the project. 😄 

I can also try building and running my own image with the fix for now, if it's not going to be released anytime soon-ish. But I figured it was worth asking before I went down that path 😆

**Comment by ahayworth** [comment]
> This isn't a high priority issue for us at the moment and I don't expect that it will be fixed in the short term. The native server is still a WIP with many features disabled etc.

@ryanemerson That's actually really good to know, thank you! I had assumed that it was more or less a feature-complete, supported alternative - but that assumption was clearly incorrect! 😆

I'll take your advice and keep working with the JVM image for now. Thank you for the response, I appreciate it!

**Pull request closed without merging**